src/google/protobuf/generated_message_tctable_full.cc

```diff
@@ -58,15 +59,22 @@ const char* TcParser::ReflectionFallback(PROTOBUF_TC_PARAM_DECL) {
   }
 
   SyncHasbits(msg, hasbits, table);
+
+  auto* full_msg = DownCast<Message*>(msg);
+
+  if ((table->extension_offset & 1) != 0) {
```

Comment:
Also curious about the rationale for this change.

Reply:
Something changed when merging with the latest versions where message set has been moved to table driven parsing as well. I incorporated this by adding this to indicate message set